File: Quiz.Task/src/Quiz.Task.Service/Models/Class.cs
using Quiz.Common.Models;

namespace Quiz.Task.Service.Models
{
    public class Class : IModel
    {
        public int Id { get; set;}
        public string ClassName { get; set;}
        public DateTime CreateDate { get; set;}
        public string Description { get; set;}
        public int UserId { get; set;}
        public virtual UserInfo User { get; set;} 
        public virtual ICollection<Folder> Folders { get; set; }
        public virtual ICollection<Member> Members { get; set; }
        
    }
}

File: quiz/lib/models/class.dart
import 'package:intl/intl.dart';
import 'package:quiz/models/user.dart';

class Class{
  int classId;
  String className;
  DateTime? createDate;
  String description;
  UserInfo user;
  int? NumOfFolder;
  int? NumOfMember;
  Class({required this.classId, required this.className, required this.description, required this.user, this.createDate, this.NumOfMember, this.NumOfFolder});

  factory Class.fromJson(Map<String, dynamic> json) => Class(
    classId: json['id'],
    className: json['className'],
    description: json['description'],
    user: UserInfo.fromJson(json['user']),
    createDate: json['createDate']!= null? DateFormat('dd-MM-yyyy').parse(json['createDate']): null,
    NumOfFolder: json['numOfFolder'],
    NumOfMember: json['numOfMember']
  );
}

class ClassRequest{
  int classId;
  String className;
  String description;

  ClassRequest({required this.classId, required this.className, required this.description});
}
